### PR TITLE
[picarto.tv] Fix regarding changed URL

### LINKED ISSUE #1932: Can't Download Picarto Video
## Bug Report

- [X ] I have read the contribution guidelines. 

### Description

Picarto.tv plugin doesn't seem to download any recorded streams anymore

### Expected / Actual behavior

Expected it to download video, didn't

### Reproduction steps / Explicit stream URLs to test

1. streamlink -o Amarynceus.flv https://picarto.tv/videopopout/Amarynceus_2018.07.04.22.55.48.flv

### Logs

```
streamlink -o Amarynceus.flv https://picarto.tv/videopopout/Amarynceus_2018.07.04.22.55.48.flv -l debug
[cli][debug] OS:         Linux-4.15.0-24-generic-x86_64-with-Ubuntu-18.04-bionic
[cli][debug] Python:     2.7.15rc1
[cli][debug] Streamlink: 0.14.2+24.g6bf654a2
[cli][debug] Requests(2.19.1), Socks(1.6.7), Websocket(0.48.0)
[cli][info] Found matching plugin picarto for URL https://picarto.tv/videopopout/Amarynceus_2018.07.04.22.55.48.flv
[plugin.picarto][debug] Possible VOD stream...
[plugin.picarto][warning] Probably a VOD stream but no VOD found?
[plugin.picarto][error] The channel Amarynceus_2018.07.04.22.55.48.flv does not exist
error: No playable streams found on this URL: https://picarto.tv/videopopout/Amarynceus_2018.07.04.22.55.48.flv
```

### PR DESCRIPTION
**Pull request by skulblakka**
Looks like picarto started escaping the slashes after the https in their VOD URL. 

This adjusts the regex accordingly and removes the slashes (I hope this is the correct way to do that).

Fixes #1932

### CODE REVIEW

File: src/streamlink/plugins/picarto.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def _create_flash_stream(self, server, channel, token):
     def _get_vod_stream(self, page):
         m = self._vod_re.search(page.text)
         if m:
-            return HLSStream.parse_variant_playlist(self.session, m.group(1))
+            return HLSStream.parse_variant_playlist(self.session, m.group(1).replace('\\/', '/'))
```
Comment (beardypig) [inline src/streamlink/plugins/picarto.py]: You could use `parse_json` if you included the `"` characters in the match group. It looks like JSON so there might be other things escaped for other streams...

**Comment by skulblakka** [comment]
@beardypig Is this what you meant? If yes, is the new regex ok this way?

**Comment by beardypig** [comment]
Yeah like that. But, I’m not sure the regex is exactly how you meant it - or I don’t understand it :) 

Should it match `#vod-player", {foo}`? I think at the moment it will match things like `v {foo}`, which might not be how you wanted it. You can check it on regex101.com, which has some nice features for debugging regex :)

**Comment by skulblakka** [comment]
>But, I’m not sure the regex is exactly how you meant it

Not unlikely. I can't say I have much experience using regular expressions.

>Should it match #vod-player", {foo}?

I want it to match `{foo}` after `#vod-player", ` (ight now you could also match `{foo}` by just matching everything between two curly braces as they this is the only occurrence in the page but I thought that would be to generic).

Looking at the explanation of the expression on regex101.com (thanks for the suggestion btw 👍 ) I see what you mean. I updated the regex hoping it does the right thing now.

**Comment by beardypig** [comment]
No problem. I’m not sure you need to escape the `-` in this case, otherwise it looks good. I believe you can change the regex engine to Python on regex101 which might highlight any unnecessary escaping, or if you use a decent IDE it’ll do the same.

**Pull request merged**